lib/ClangImporter/ImportDecl.cpp

```diff
@@ -7904,187 +7904,216 @@ createUnavailableDecl(Identifier name, DeclContext *dc, Type type,
+  }
+}
 
-
+void ClangImporter::Implementation::loadAllMembersOfObjcContainer(Decl *D, const clang::ObjCContainerDecl *objcContainer) {
   clang::PrettyStackTraceDecl trace(objcContainer, clang::SourceLocation(),
                                     Instance->getSourceManager(),
                                     "loading members for");
-
   DeclContext *DC;
   IterableDeclContext *IDC;
-  SmallVector<ProtocolDecl *, 4> protos;
+  {
+    ExtensionDecl *ext;
+    figureOutTheDeclarationContextToImportInto(D, ext, DC, IDC);
+    if (ext)
+      loadMembersOfBaseImportedFromClang(ext);
+  }
+  
+  ImportingEntityRAII Importing(*this);
+  
+  SmallVector<Decl *, 16> members = collectMembersToAdd(objcContainer, D, DC);
+  
+  // Add the members now, before ~ImportingEntityRAII does work that might
+  // involve them.
+  for (auto member : members) {
+    IDC->addMember(member);
+  }
+}
 
-  // Figure out the declaration context we're importing into.
+static void figureOutTheDeclarationContextToImportInto(Decl* D, ExtensionDecl *&ext, DeclContext *&DC, IterableDeclContext *&IDC) {
   if (auto nominal = dyn_cast<NominalTypeDecl>(D)) {
+    ext = nullptr;
     DC = nominal;
     IDC = nominal;
-  } else {
-    auto ext = cast<ExtensionDecl>(D);
+  }
+  else {
+    ext = cast<ExtensionDecl>(D);
     DC = ext;
     IDC = ext;
+  }
+}
 
 
-    // If the base is also imported from Clang, load its members first.
-    const NominalTypeDecl *base = ext->getExtendedType()->getAnyNominal();
-    if (auto *clangBase = base->getClangDecl()) {
-      base->loadAllMembers();
-
-      // Sanity check: make sure we don't jump over to a category /while/
-      // loading the original class's members. Right now we only check if this
-      // happens on the first member.
-      if (auto *clangContainer = dyn_cast<clang::ObjCContainerDecl>(clangBase)){
-        if (!clangContainer->decls_empty()) {
-          assert(!base->getMembers().empty() &&
-                 "can't load extension members before base has finished");
-        }
-      }
+static void loadMembersOfBaseImportedFromClang(ExtensionDecl *ext) {
+  const NominalTypeDecl *base = ext->getExtendedType()->getAnyNominal();
+  auto *clangBase = base->getClangDecl();
+  if (!clangBase)
+    return;
+  base->loadAllMembers();
+  
+  // Sanity check: make sure we don't jump over to a category /while/
+  // loading the original class's members. Right now we only check if this
+  // happens on the first member.
+  if (auto *clangContainer = dyn_cast<clang::ObjCContainerDecl>(clangBase)){
+    if (!clangContainer->decls_empty()) {
+      assert(!base->getMembers().empty() &&
+             "can't load extension members before base has finished");
     }
   }
+}
 
-  ImportingEntityRAII Importing(*this);
+void ClangImporter::Implementation::insertMembersAndAlternates(const clang::NamedDecl *nd, SmallVector<Decl *, 16> &members) {
+  llvm::SmallPtrSet<Decl *, 4> knownAlternateMembers;
+  forEachDistinctName(nd,
```

Comment:
Can you indent this like code, ala the LLVM coding standards:

https://llvm.org/docs/CodingStandards.html#format-lambdas-like-blocks-of-code